Source/Test/SystematicTesting/TestingEngine.cs

```diff
@@ -267,7 +267,8 @@ public void Run()
 
             if (this.Configuration.IsTelemetryEnabled)
             {
-                this.TrackTelemetry();
+                // 10 seconds should be enough, any more than that and too bad.
+                this.TrackTelemetry().Wait(10000, this.CancellationTokenSource.Token);
```

Comment:
What is the typical wait time btw?

Also, I wonder if we should check (not here, probably inside the client) if there is even network connection and if not to just disable telemetry and return a Completed.Task? (This could optimize cases where say someone is for some reason having left telemetry on, and run tests in some no-network box.) Perhaps this is already implemented, I have not looked into the actual underlying logic yet :)

Reply:
A quick unscientific local test is getting round trip times in the 50-70 millisecond range.  Google is very fast with all their services.  When I was playing with firebase I was getting round trip times in the under 10 ms range sometimes down to 2 ms which is simply nuts, I was calculating it was basically the speed of light... let's say I was lucky and were hitting their Dalles Oregon datacenter, that's 267 miles, round trip 534, which in 3ms is /0.003 is 178,000 miles per second. Speed of light is 186,000 mi/s!

Note the [Google Analytics GA4 SLA](https://marketingplatform.google.com/about/analytics_products/sla/) talks about somewhere upward of 96%, so it is possible a call also fail when GA4 is down also and it's hard to know how much retry logic happens automatically in an HTTP stack which is why I put a 10 second cap on it.

Good idea checking for network availability.   Checking network availability is a bit tricky sometimes, because you can have a network interface that is talking to a local router, but no internet connection, and the only way to really check for an internet connection is to try ping something outside, well if we're doing that then we might as well make this the ping!

There is `System.Net.NetworkInformation.NetworkInterface.GetIsNetworkAvailable()` but guess what, it returns true when I put my machine into airplane mode :-)